src/features/utils/instrument-base.js

```diff
@@ -52,36 +51,45 @@ export class InstrumentBase extends FeatureBase {
     this.onAggregateImported = undefined
 
     /** used in conjunction with newrelic.start() to defer harvesting in features */
-    if (agentRef.init[this.featureName].autoStart === false) this.auto = false
-    /** if the feature requires opt-in (!auto-start), it will get registered once the api has been called */
-    if (this.auto) registerDrain(agentRef.agentIdentifier, featureName)
-    else {
-      this.ee.on('manual-start-all', single(() => {
+    this.deferred = Promise.resolve()
+
+    if (!auto || agentRef.init[this.featureName].autoStart === false) {
+      this.deferred = new Promise((resolve, reject) => {
+        this.ee.on('manual-start-all', single(() => {
         // register the feature to drain only once the API has been called, it will drain when importAggregator finishes for all the features
         // called by the api in that cycle
-        registerDrain(agentRef.agentIdentifier, this.featureName)
-        this.auto = true
-        this.importAggregator(agentRef)
-      }))
+          registerDrain(agentRef.agentIdentifier, this.featureName)
+          resolve()
+        }))
+      })
+    } else {
+      /** if the feature requires opt-in (!auto-start), it will get registered once the api has been called */
+      registerDrain(agentRef.agentIdentifier, featureName)
```

Comment:
I don't quite get the need for this change

Reply:
IMO it makes it a little simpler to not need to mock out auto and pass it around, and makes it so that importAggregator is always called in just one place.  Just wait for the deferment promise to resolve, it either does so immediately if auto, or after the start api if not

Reply:
Yeah nvm, I think cutting down importAggregator to 1 call makes it simpler right.